pkg/util/template/funcerror.go

```diff
@@ -0,0 +1,64 @@
+package template
+
+import (
+	"github.com/authgear/authgear-server/pkg/api/apierrors"
+)
+
+// This function takes an api error and returns a non-validation error
+//
+// i.e. err.Reason != `ValidationFailed`
+//
+// otherwise it returns nil
+func GetNonValidationError(apiErr *apierrors.APIError) (nonValidationErr *apierrors.APIError) {
+	if apiErr == nil {
+		return nil
+	}
+	switch apiErr.Reason {
+	case "ValidationFailed":
+		return nil
+	default:
+		return apiErr
+	}
+}
+
+// This function takes any error and returns a validation error
+//
+// i.e. err.Reason == `ValidationFailed`
+//
+// otherwise it returns nil
+func GetValidationError(apiErr *apierrors.APIError, inputFieldName string) (validationErr *apierrors.APIError) {
```

Comment:
This is not correct. What I told you is to return a validation error with details specific to the given field. So this function should
1. Clone apiErr
2. Filter out any details that is not related to inputFieldName.
3. Return the new apiErr

Reply:
> Filter out any details that is not related to inputFieldName.

Say given below error and inputFieldName `password_input`, by `filter out any details`, should we remove all other items in `missing`?

## Input apiErr

```
{
	"code": 400,
	"info": {
		"causes": [
			{
				"details": {
					"actual": [
						"gorilla.csrf.Token",
						"x_step"
					],
					"expected": [
						"x_confirm_password",
						"x_password"
					],
					"missing": [
						"x_confirm_password",
						"x_password"
					]
				},
				"kind": "required",
				"location": ""
			}
		]
	},
	"message": "invalid value",
	"name": "Invalid",
	"reason": "ValidationFailed"
}
```

## output

```
{
	"code": 400,
	"info": {
		"causes": [
			{
				"details": {
					"actual": [
						"gorilla.csrf.Token",
						"x_step"
					],
					"expected": [
						"x_confirm_password",
						"x_password"
					],
					"missing": [
						"x_password"
					]
				},
				"kind": "required",
				"location": ""
			}
		]
	},
	"message": "invalid value",
	"name": "Invalid",
	"reason": "ValidationFailed"
}
```

Reply:
Good question. This is a special case. The keyword `required` reports the error in location one level up. So you're right in this case.

Please also add tests to test this filter function,

Reply:
2 cases
- Normal case - `location` is non-empty, can just use
- special case - `location` is empty, need to loop through `missing`

```
`/login_id` -> phone
`/x_login_id` -> phone
```

### Proposed api

```
{{ getValidationError apiErr "login_id" "x_login_id" }}
```